all-modules/build.gradle

```diff
@@ -1,9 +1,10 @@
 // This module depends on all other Venice modules. Its primary purpose is to help scripts identify all venice modules
 // via transitive dependencies instead of enumerating all modules individually.
 dependencies {
-    rootProject.subprojects.each { subproject ->
-        if (subproject.path != project.path && subproject.subprojects.size() == 0) {
-            implementation project(subproject.path)
-        }
+  rootProject.subprojects.each { subproject ->
+    if (subproject.path != project.path && subproject.subprojects.isEmpty()) {
+      implementation project(subproject.path)
+      implementation project(path: ':internal:venice-test-common', configuration: 'integrationTestUtils')
```

Comment:
Is it possible to make this more generic? If there is an `integrationTestUtils` configuration in the subproject, then depend on that by default

In the current form of this PR, does this `implementation` need to be specified in the loop? Shouldn't this be moved directly under `dependencies` section?

Reply:
Is it possible to make this module depend on all configurations that generate an artifact? I think we will always forget updating this module and realize only later

Reply:
Is it possible to make this module depend on all configurations that generate an artifact? I think we will always forget updating this module and realize only later

Reply:
This will require more complex Gradle code and create a risk of pulling something that we don't need/want and having duplicate classes as a result. So I prefer to keep it as is.